### PR TITLE
Antiautodefenestration

### PR DESCRIPTION
**Pull request by tylershuster**
don't do it @galenwp 

(two commits preventing accidental deletion of data — feel free to suggest more)

### CODE REVIEW

**Comment by galenwp** [comment]
life saving

File: pkg/interface/src/views/apps/publish/components/MarkdownEditor.tsx [hunk 1]
```diff
@@ -22,6 +24,17 @@ interface MarkdownEditorProps {
   onBlur?: (e: any) => void;
 }
 
+const PromptIfDirty = () => {
+  const formik = useFormikContext();
+  usePreventWindowUnload(formik.dirty);
+  return (
+    <Prompt
+      when={formik.dirty}
+      message="Are you sure you want to leave? You have with unsaved changes."
```
Comment (lycurgus) [inline pkg/interface/src/views/apps/publish/components/MarkdownEditor.tsx:33]: Tiny nit: "have unsaved" ?

File: pkg/interface/src/logic/lib/util.js [hunk 2]
```diff
@@ -325,3 +326,12 @@ export function numToUd(num) {
     f.join('.')
   )(num.toString())
 }
+
+export function usePreventWindowUnload(shouldPreventDefault) {
+  React.useEffect(() => {
+    if (!shouldPreventDefault) return;
+    const handleBeforeUnload = event => event.preventDefault();
```
Comment (matildepark) [inline pkg/interface/src/logic/lib/util.js:333]: This doesn't work on Chrome for whatever reason, you can still close the tab.

Comment (tylershuster) [reply pkg/interface/src/logic/lib/util.js]: Looks like Chrome uses a slightly different approach. Give this a shot.

**Review by matildepark — APPROVED** [review]

**Comment by matildepark** [comment]
lol @tylershuster this is a little too aggressive — fires a prompt if you submit a post normally

**Comment by tylershuster** [comment]
Oh ha. hold plz

**Pull request merged**